docs/internals-mappings.md

```diff
@@ -0,0 +1,428 @@
+
+Manual Field Mappings allow the use of custom logic to map the value of a Field within the Terraform Schema onto a Field within an SDK Model.
+
+As such Manual Field Mappings likely need to be structured similarly to the other types of Field Mapping, but also allow for two mapping functions to be defined (to map bidirectionally) - as a hypothetical example:
+
+```json
+{
+  "type": "Manual",
+  "manualMapping": {
+    "schemaModelName": "VirtualMachineResourceSchema",
+    "schemaFieldPath": "SomeComplexProperty",
+    "schemaToSdkMethodName": "mapSomeComplexPropertyToSomePayloadProperty",
+    "sdkModelName": "VirtualMachineProperties",
+    "sdkFieldPath": "SomePayloadProperty",
+    "sdkToSchemaMethodName": "mapSomePayloadPropertyToSomeComplexProperty"
+  }
+}
+```
+
+In this example the following properties would exist:
+
+* `schemaModelName` - specifies the name of the Terraform Schema Model where the Field that should be mapped to/from exists.
+* `schemaFieldPath` - specifies the path to the Field within the Terraform Schema Model that should be mapped to/from the Field within the SDK.
+* `schemaToSdkMethodName` - specifies the name of a function that should be assumed to exist within the Terraform Provider, which maps the value of the Terraform Schema Field onto the SDK Field.
+* `sdkModelName` - specifies the name of the SDK Model where the SDK Field defined in `sdkFieldPath` exists.
+* `sdkFieldPath`- specifies the path to the Field within the SDK Model where the Schema Field should be mapped to/from.
+* `sdkToSchemaMethodName` - specifies the name of a function that should be assumed to exist within the Terraform Provider, which maps the value of the SDK Field onto the Terraform Schema Field.
+
+> **Note:** The current implementation that exists exposed only `methodName` - as such is only useful for one-way value transformations.
+
+## 2: Model to Model Mappings
+
+A ModelToModel mapping defines that a Terraform Schema Model should be mapped onto an SDK Model.
+
+> **Note:** A ModelToModel Mapping and a ModelToModel Field Mapping are related but different - a ModelToModel Field Mapping _must_ have an associated ModelToModel Mapping - however a ModelToModel mapping can exist without a ModelToModel Field Mapping.
+
+Model to Model Mappings support the following properties:
+
+* `schemaModelName` - specifies the name of the Terraform Schema Model that the SDK Model should be mapped to/from.
+* `sdkModelName` - specifies the name of the SDK Model that the Terraform Schema Model should be mapped to/from.
+
+An example of a Model to Model mapping can be seen below:
+
+```json
+{
+  "modelToModelMappings": [
+    {
+      "schemaModelName": "VirtualMachineResourceSchema",
+      "sdkModelName": "VirtualMachine"
+    }
+  ]
+}
+```
+
+> **Note:** An end-to-end example can be found under `Example` below.
```

Comment:
I understand this is trying to be helpful but repeating it multiple times throughout this guide is unnecessary bloat and noise for a topic that's complex. Say it once, probably at the top of the document.